CMakeLists.txt

```diff
@@ -72,6 +72,14 @@ target_compile_definitions(mbed-core
         ${MBED_CONFIG_DEFINITIONS}
 )
 
+# Add MBED_TEST_MODE for Greentea testing backward compatibility with CLI1
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
```

Comment:
`BUILD_TESTING` is set by CTest, which is used by unit tests only. If I understand correctly, Greentea tests don't use CTest - they are normal Mbed applications with [`greentea-client`](https://github.com/ARMmbed/mbed-os/tree/master/features/frameworks/greentea-client) used as a library, to allow synchronisation/reporting between host and target.

See this Greentea test CMake definition for example: https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.9.0/platform/tests/TESTS/mbed_platform/wait_ns/CMakeLists.txt. CTest is not used.

So I'm not sure this would work for a real Greentea tests.

Reply:
Greentea tests require more work. They will use CTest.

As they are now, we should explicitly enable `BUILD_TESTING`, adding into the function `mbed_greentea_add_test()` ?

Reply:
This would be a separete discussion, but I'd like to have a rough idea how CTest will be used with Greentea. If it'll be used for all test suites, then it makes sense to add it to `mbed_greentea_add_test()`.